src/index.d.ts

```diff
@@ -9,54 +9,56 @@ export interface RoutineTypes {
     FULFILL: string;
 }
 
-export type RoutineAction<T> = Action<T>;
-
-export type RoutineActionCreator<T> = (payload?: T) => RoutineAction<T>;
-
-export interface RoutineActionCreators {
-    trigger: RoutineActionCreator<any>;
-    request: RoutineActionCreator<any>;
-    success: RoutineActionCreator<any>;
-    failure: RoutineActionCreator<any>;
-    fulfill: RoutineActionCreator<any>;
+export interface RoutineActionCreators<TActionCreator> {
+    trigger: TActionCreator;
```

Comment:
The `trigger` action creator is strongly-typed if called as `myRoutine(typedPayload)`, so it should be consistent with `myRoutine.trigger(typedPayload)`